activerecord/lib/active_record/validations/uniqueness.rb

```diff
@@ -86,12 +86,21 @@ def build_relation(klass, attribute, value) # :nodoc:
       def scope_relation(record, relation)
         Array(options[:scope]).each do |scope_item|
           if reflection = record.class._reflect_on_association(scope_item)
```

Comment:
I think it's a good suggestion too.  

The updated version
```ruby
scope_value = if record.class._reflect_on_association(scope_item)
  record.association(scope_item).reader
else
  record._read_attribute(scope_item)
end
```
makes more clear what's going on. Compare with the previous version:  
```ruby
scope_value = if record.class._reflect_on_association(scope_item)
  record.send(scope_item)
else
  record._read_attribute(scope_item)
end
```